dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java

```diff
@@ -92,24 +120,318 @@ public ContainerResource(final WebResource webResource,
+
+            return Response.ok(response)
+                .build();
+
+
+        }
+
+        ShortyId containerShorty = APILocator.getShortyAPI()
+            .getShorty(containerId)
+            .orElseGet(() -> {
+                throw new ResourceNotFoundException("Can't find Container:" + containerId);
+            });
+
+        Container container = (containerShorty.subType == ShortType.CONTAINER) ? APILocator.getContainerAPI()
+            .find(containerId, user, mode == PageMode.ANON)
+                : (mode.showLive) ? (Container) APILocator.getVersionableAPI()
+                    .findLiveVersion(containerShorty.longId, user, mode == PageMode.ANON)
+                        : (Container) APILocator.getVersionableAPI()
+                            .findWorkingVersion(containerShorty.longId, user, mode == PageMode.ANON);
+
+
+        Identifier containerIdentifier = APILocator.getIdentifierAPI()
+            .find(container);
+
+        org.apache.velocity.context.Context context = VelocityUtil.getWebContext(req, res);
+        context.remove("EDIT_MODE");
+
+
+
+        context.put("contentletList" + container.getIdentifier(), Lists.newArrayList(contentlet.getIdentifier()));
+        StringWriter out = new StringWriter();
+
+        final ContainerStructure cStruct = APILocator.getContainerAPI()
+            .getContainerStructures(container)
+            .stream()
+            .filter(cs -> contentlet.getStructureInode()
+                .equals(cs.getStructureId()))
+            .findFirst()
+            .orElseGet(() -> {
+                throw new ResourceNotFoundException("Can't find container structure template:" + contentletId);
+            });
+
+
+
+        // defensive copy
+        container = (Container) BeanUtils.cloneBean(container);
+        container.setPreLoop(null);
+        container.setPostLoop(null);
+
+
+        VelocityUtil.getEngine()
+            .evaluate(context, out, this.getClass()
+                .getName(), ContainerServices.buildVelocity(container, containerIdentifier, false));
+
+        Map<String, String> response = new HashMap<>();
+        response.put("render", out.toString());
+
+        return Response.ok(response)
+            .build();
+
+    }
+
+
+
+    @POST
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("add/{containerId}/content/{contentletId}/uid/{uid}/order/{order}")
+    public final Response addContentToContainer(@Context final HttpServletRequest req, @Context final HttpServletResponse res,
+            @PathParam("containerId") final String containerId, @PathParam("contentletId") final String contentletId,
+            @QueryParam("order") final int order, @PathParam("uid") final String uid) throws DotDataException,
+            DotSecurityException, ParseErrorException, MethodInvocationException, ResourceNotFoundException, IOException,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.velocity.exception.ParseErrorException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.velocity.exception.MethodInvocationException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.velocity.exception.ResourceNotFoundException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)